### PR TITLE
Add gateway online/offline events

### PR DESCRIPTION
**Pull request by DmitrySamoylov**
Currently `lumeod` sends it's offline/online status in some uncommon format. So the plan is to change it to JSON just like all other events. To achieve it I add a (de)serializable `gateway::Event` type here.

### CODE REVIEW

File: events/src/gateway.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use serde::{Deserialize, Serialize};
+
+pub type Version = String;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Event {
+    Online(Version),
+    Offline,
+}
```
Comment (jplatte) [inline events/src/gateway.rs:9]: The default serialization for this is going to produce `{ "Online": "<version>" }` for online and `{ "Offline": null }` AFAICT.

I would much prefer internal tagging plus the version being a struct variant, so get `{ "status": "online", "version": "<version>" }` and `{ "status": "offline" }`:

```suggestion
#[serde(tag = "status", rename_all = "lowercase")]
pub enum Event {
    Online { version: String },
    Offline,
}
```

File: events/src/gateway.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+use serde::{Deserialize, Serialize};
+
+pub type Version = String;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(tag = "status", rename_all = "lowercase")]
+pub enum Event {
+    Online { version: String },
```
Comment (zeenix) [inline events/src/gateway.rs:8]: you forgot to your type alias. :)

**Review by zeenix — APPROVED** [review]
LGTM otherwise.

File: events/src/gateway.rs (hunk 2 above)
Comment (DmitrySamoylov) [reply events/src/gateway.rs:8]: > you forgot to your type alias. :)

Not sure but looks like you forgot a verb :) Can you rephrase?

Comment (zeenix) [reply events/src/gateway.rs:8]: > > you forgot to your type alias. :)
> 
> Not sure but looks like you forgot a verb :) Can you rephrase?

lol. "use" is the verb. You used `String`.

Comment (DmitrySamoylov) [reply events/src/gateway.rs:8]: Ah, yeah, I removed it because I used it to show that this String in an unnamed field is actually a version but now the field is named thanks to @jplatte 's suggestion.

Comment (jplatte) [reply events/src/gateway.rs:8]: I think the typedef can then be removed, right?

Comment (DmitrySamoylov) [reply events/src/gateway.rs:8]: Yeah, it is removed.

**Review by jplatte — APPROVED** [review]
